Integration: Add page for session persistence using Firestore for ADK Java (#1563)
google/adk-docs: docs/integrations/firestore-session-service.md

---

> catalog_title: Session Management with Firestore

Reviewer: ```suggestion
catalog_title: Firestore Session Service
```

> catalog_description: Session state management for ADK agents using Google Cloud Firestore

Reviewer: ```suggestion
catalog_description: Session state management for ADK agents using Firestore
```

catalog_icon: /integrations/assets/firestore-session.jpg
catalog_tags: ["data", "google"]
---

> # Session State Management for ADK agents using Google Cloud Firestore

Reviewer: ```suggestion
# Session State Management using Firestore
```

## Use cases

- **Customer Support Agents**: Maintain context across long-running support tickets, allowing the agent to remember past troubleshooting steps and preferences across multiple sessions.
- **Personalized Assistants**: Build agents that accumulate knowledge about the user over time, personalizing future interactions based on historical conversations.
- **Multi-modal Workflows**: Seamlessly handle complex use cases involving images, videos, and audio alongside text conversations, leveraging the built-in GCS artifact storage.
- **Enterprise Chatbots**: Deploy highly reliable, conversational AI applications with production-grade persistence suitable for large-scale enterprise environments.

## Prerequisites

- A [Google Cloud Project](https://cloud.google.com/) with Firestore enabled
- A [Firestore database](https://cloud.google.com/firestore/docs/setup) in your Google Cloud Project
- Appropriate [Google Cloud credentials](https://cloud.google.com/docs/authentication/provide-credentials-adc) configured in your environment

## Install dependencies

!!! note

>    Ensure you use the same version for both `google-adk` and `google-adk-firestore-session-service` to guarantee compatibility.

Reviewer: ```suggestion
    Ensure you use the same version for both `google-adk` and `google-adk-firestore-session-service` to guarantee compatibility.
```

>     The `google-adk-firestore-session-service` version matches the core Google Agent ADK version. Ensure you use the same version for both libraries to guarantee compatibility.

Reviewer: ```suggestion
    Ensure you use the same version for both `google-adk` and `google-adk-firestore-session-service` to guarantee compatibility.
```

            .build();
    }
}
```

## Configuration

!!! note

    The Firestore Session Service supports properties file configuration. This allows you to easily target a dedicated Firestore database and define custom collection names for storing your agent session data.

You can customize your ADK application to use the Firestore session service by providing your own Firestore property settings, otherwise the library will use the default settings. 

### Environment-Specific Configuration

The library prioritizes environment-specific property files over the default settings using the following resolution order:

1. **Environment Variable Override**: It first checks for an environment variable named `env`. If this variable is set (e.g., `env=dev`), it will attempt to load a properties file matching the template: `adk-firestore-{env}.properties` (e.g., `adk-firestore-dev.properties`).
2. **Default Fallback**: If the `env` variable is not set, or the environment-specific file cannot be found, the library defaults to loading `adk-firestore.properties`.

Sample Property Settings:

```properties
# Firestore collection name for storing session data

> adk.firestore.collection.name=adk-session

Author: you are correct @koverholt , i had this corrected  these keys now and also made amend to the  [https://github.com/google/adk-java/pull/1096 ](url) as well.
